static/js/compose_actions.js

```diff
@@ -215,6 +215,11 @@ exports.start = function (msg_type, opts) {
         opts.private_message_recipient = '';
     }
 
+    var subbed_streams = stream_data.subscribed_subs();
+    if (subbed_streams.length === 1 && (opts.trigger === "new topic button" || opts.trigger === "compose_hotkey" && msg_type === "stream")) {
+        opts.stream = subbed_streams[0].name;
+    }
+
```

Comment:
This seems worth having a unit test for; can you add one?  

I think the way to do this cleanly is (1) A commit that renames `fill_in_opts_from_current_narrowed_view` to `exports._fill_in_opts_defaults`, including moving the block above this one to it, (2) a commit adding a basic unit test suite for that relatively pure data manipulation function, and (3) a commit adding this new check and tests for it.

Can you do that?

Reply:
Hmm, first you might want to look at the existing `compose_actions.js` node tests for `start`; possibly we've already done the test setup nicely and don't need to do that refactoring.